lib/rubocop/cop/style/access_modifier_declarations.rb

```diff
@@ -130,6 +158,54 @@ def message(range)
             format(INLINE_STYLE_MESSAGE, access_modifier: access_modifier)
           end
         end
+
+        def find_corresponding_def_node(node)
+          if access_modifier_with_symbol?(node)
+            method_name = node.arguments.first.value
+            node.parent.each_child_node.find do |child|
+              child.def_type? && child.method?(method_name)
```

Comment:
`each_child_node` can filter node types.

```suggestion
            node.parent.each_child_node(:def).find do |child|
              child.method?(method_name)
```